### DISCUSSION TITLE
Airflow seems to be spamming Postrges RDS Error Log

### DISCUSSION THREAD

**Discussion by mldedalus** [apache/airflow#34084] [Q&A]
We have deployed Airflow 2.6.3 to AWS using the latest Helm charts.
Everything works - can see no issues in pod logs.

However, we quickly ran out of space in our Postgres RDS instance. There are roughly 10Gb of error logs being generated per hour, and it looks like this - looks very much like a low level SQL logging?

Any idea what could be causing this?

We have other apps on our RDS instance and they dont appear to have the same issue

```console
2023-09-04 15:44:38 UTC:172.16.1.144(35226):airflow@airflow:[24579]:LOG:  statement: SELECT 1
2023-09-04 15:44:38 UTC:172.16.1.144(35226):airflow@airflow:[24579]:LOG:  plan:
2023-09-04 15:44:38 UTC:172.16.1.144(35226):airflow@airflow:[24579]:DETAIL:     {PLANNEDSTMT 
	   :commandType 1 
	   :queryId 1147616880456321454 
	   :hasReturning false 
	   :hasModifyingCTE false 
	   :canSetTag true 
	   :transientPlan false 
	   :dependsOnRole false 
	   :parallelModeNeeded false 
	   :jitFlags 0 
	   :planTree 
	      {RESULT 
	      :startup_cost 0.00 
	      :total_cost 0.01 
	      :plan_rows 1 
	      :plan_width 4 
	      :parallel_aware false 
	      :parallel_safe false 
	      :async_capable false 
	      :plan_node_id 0 
	      :targetlist (
	         {TARGETENTRY 
	         :expr 
	            {CONST 
	            :consttype 23 
	            :consttypmod -1 
	            :constcollid 0 
	            :constlen 4 
	            :constbyval true 
	            :constisnull false 
	            :location 7 
	            :constvalue 4 [ 1 0 0 0 0 0 0 0 ]
	            }
	         :resno 1 
	         :resname ?column? 
	         :ressortgroupref 0 
	         :resorigtbl 0 
	         :resorigcol 0 
	         :resjunk false
	         }
	      )
	      :qual <> 
	      :lefttree <> 
	      :righttree <> 
	      :initPlan <> 
	      :extParam (b)
	      :allParam (b)
	      :resconstantqual <>
	      }
	   :rtable (
	      {RANGETBLENTRY 
	      :alias <> 
	      :eref 
	         {ALIAS 
	         :aliasname *RESULT* 
	         :colnames <>
	         }
	      :rtekind 8 
	      :lateral false 
	      :inh false 
	      :inFromCl false 
	      :requiredPerms 0 
	      :checkAsUser 0 
	      :selectedCols (b)
	      :insertedCols (b)
	      :updatedCols (b)
	      :extraUpdatedCols (b)
	      :securityQuals <>
	      }
	   )
	   :resultRelations <> 
	   :appendRelations <> 
	   :subplans <> 
	   :rewindPlanIDs (b)
	   :rowMarks <> 
	   :relationOids <> 
	   :invalItems <> 
	   :paramExecTypes <> 
	   :utilityStmt <> 
	   :stmt_location 0 
	   :stmt_len 0
	   }
	
2023-09-04 15:44:38 UTC:172.16.1.144(35226):airflow@airflow:[24579]:STATEMENT:  SELECT 1
2023-09-04 15:44:38 UTC:172.16.1.144(35226):airflow@airflow:[24579]:LOG:  statement: BEGIN
2023-09-04 15:44:38 UTC:172.16.1.144(35226):airflow@airflow:[24579]:LOG:  statement: SELECT dag_schedule_dataset_reference.dag_id AS dag_schedule_dataset_reference_dag_id, max(dataset_dag_run_queue.created_at) AS last_queued_time, min(dataset_dag_run_queue.created_at) AS first_queued_time 
	FROM dag_schedule_dataset_reference LEFT OUTER JOIN dataset_dag_run_queue ON dag_schedule_dataset_reference.dataset_id = dataset_dag_run_queue.dataset_id AND dag_schedule_dataset_reference.dag_id = dataset_dag_run_queue.target_dag_id GROUP BY dag_schedule_dataset_reference.dag_id 
	HAVING count(*) = sum(CASE WHEN (dataset_dag_run_queue.target_dag_id IS NOT NULL) THEN 1 ELSE 0 END)
2023-09-04 15:44:38 UTC:172.16.1.144(35226):airflow@airflow:[24579]:LOG:  plan:
2023-09-04 15:44:38 UTC:172.16.1.144(35226):airflow@airflow:[24579]:DETAIL:     {PLANNEDSTMT 
	   :commandType 1 
	   :queryId 18299031726445585787 
	   :hasReturning false 
	   :hasModifyingCTE false 
	   :canSetTag true 
	   :transientPlan false 
	   :dependsOnRole false 
	   :parallelModeNeeded false
```

**Comment by potiuk** [comment]
Look for all the ways you can enable `log_statement = all`- driver configuration for example. Airflow on it's own does not enable it but likely either your postgres instance/schema has it turned on (for example when someone debugged it) or your driver configuration has it set (see for all the configuration parameters in https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#database) 

Also - Airlfow will generate a lot of queries - it uses the DB for scheduling and makes a lot of complex queries. Maybe your instance has the "statement logging" turned on by default, and other apps simply have many time less queries.

**Reply by Taragolis** [reply]
@potiuk A bit quicker than me 🤣 . 

Just in addition, It's a quite a difficult to configure incidentally max level of logging, e.g. at running Postgres Instance it could be done only for real superuser session, and I'm not sure that RDS superuser have enough permission for run ALTER SYSTEM, so one viable way is set
- Create RDS parameter groups
- Attach to RDS
- Restart RDS instance (optional) Not sure that all parameters could be applied without restart

So as recommendation check `log_` settings and turn it off if not required or reduce verbosity level and etc, about configuration Postgres on RDS better ask on AWS / Postgres specific resources, however this links might be useful
- https://www.postgresql.org/docs/current/runtime-config-logging.html
- https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_LogAccess.Concepts.PostgreSQL.html

BTW, if @mldedalus  use logging for metrics and performance dashbords (pgBadger for example) you could have a look [Performance Insights](https://aws.amazon.com/rds/performance-insights/) as alternative.

**Comment by Taragolis** [comment]
> Any idea what could be causing this?
> Looks very much like a low level SQL logging?

~I assume that logs turned on maximum verbosity on your RDS instance. Airflow could create a lot of queries to backend and if turn on logging everything then you could easily went out of the space.~ see https://github.com/apache/airflow/discussions/34084#discussioncomment-6907147
